test/tricks.spec.js

```diff
@@ -10,40 +10,55 @@ var tricks = {
 describe('tricks', function() {
   describe('debounce', function() {
     var spy;
+    var setTimeout = window.setTimeout;
 
     beforeEach(function() {
-      spyOn(window, 'setTimeout');
-      spy = jasmine.createSpy('debounced?');
+      jest.spyOn(window, 'setTimeout');
+      spy = jest.fn();
+    });
+
+    afterEach(function() {
+      window.setTimeout = setTimeout;
+      jest.clearAllMocks();
     });
 
     it('test debounced functions.', function() {
       var fn = debounce(spy, 50);
       fn();
 
-      expect(window.setTimeout).toHaveBeenCalledWith(jasmine.any(Function), 50);
-      window.setTimeout.calls.argsFor(0)[0]();
+      expect(window.setTimeout).toHaveBeenCalledWith(expect.any(Function), 50);
+      window.setTimeout.mock.calls[0][0]();
       expect(spy).toHaveBeenCalled();
     });
 
     it('debounced function can accept parameters', function() {
       var fn;
 
-      window.setTimeout.and.callFake(function(func) {
+      window.setTimeout.mockImplementation(function(func) {
         func();
       });
 
       fn = debounce(spy);
       fn('hello world!');
 
-      expect(spy.calls.argsFor(0)).toEqual(['hello world!']);
+      expect(spy).toHaveBeenCalledWith('hello world!');
     });
   });
 
   describe('throttle', function() {
-    var spy;
+    var spy, mockFunc;
 
     beforeEach(function() {
-      spy = jasmine.createSpy('throttled?');
+      mockFunc = jest.fn();
+
+      spy = function() {
+        var args = Array.prototype.slice.call(arguments);
+        mockFunc.apply(null, args);
+      };
```

Comment:
이 스파이는 어떤 용도일까요?

Reply:
이 스파이는 제가 `debounce` 테스트에서 `setTimeout`을 mocking 할 때 테스트 간 의존이 있다는 것을 파악하지 못하고 다른 원인에 의해서 그런것 이라 생각해서 위 처럼 스파이를 설정 했었습니다.(실제로 스파이를 위처럼 설정 후 2개의 테스트가 통과 했었습니다...)
```suggestion
      spy = jest.fn();
```
현재는 테스트간 의존성을 해결한 상태이기 때문에 이처럼 작성할 필요가 없기 때문에 위 처럼 수정하도록 하겠습니다.